gandalf.js

```diff
@@ -1,36 +1,42 @@
 /**
```

Comment:
SCRIPT-I'm a bit concerned about the potential performance impact of this loop. Have you thought about optimizing it?